app/assets/javascripts/init-elements.js

```diff
@@ -1,3 +1,65 @@
+this.Facebook = (function() {
+  var eventsBound, rootElement;
+
+  function Facebook() {}
+
+  rootElement = null;
+
+  eventsBound = false;
+
+  Facebook.load = function() {
+    if(document.getElementById('fb-root')){
+      var facebookScript, firstScript, initialRoot;
+      if (!(document.getElementById('fb-root').size() > 0)) {
+        initialRoot = document.createElement('div');
+        initialRoot.setAttribute("id", "fb-root");
+        document.body.prepend(initialRoot);
+      }
+      if (!(document.getElementById('facebook-jssdk').size() > 0)) {
+        facebookScript = document.createElement("script");
+        facebookScript.id = 'facebook-jssdk';
+        facebookScript.async = 1;
+        facebookScript.src = "//connect.facebook.net/en_US/sdk.js#xfbml=1&appId=" + (Facebook.appId()) + "&version=v2.0";
+        firstScript = document.getElementsByTagName("script")[0];
+        firstScript.parentNode.insertBefore(facebookScript, firstScript);
+      }
+      if (!Facebook.eventsBound) {
+        return Facebook.bindEvents();
+      }
+    }
+  };
+
+  Facebook.bindEvents = function() {
+    if (typeof Turbolinks !== 'undefined' && Turbolinks.supported) {
+      $(document).on('page:fetch', Facebook.saveRoot).on('page:change', Facebook.restoreRoot).on('page:load', function() {
+        return typeof FB !== "undefined" && FB !== null ? FB.XFBML.parse() : void 0;
+      });
+    }
+    return Facebook.eventsBound = true;
+  };
+
+  Facebook.saveRoot = function() {
+    return Facebook.rootElement = document.getElementById('fb-root').detach();
+  };
+
+  Facebook.restoreRoot = function() {
+    if (document.getElementById('fb-root').length > 0) {

```

Comment:
What is the length property?  `getElementById` returns a singular element or `null`